### PR TITLE
fix: hs-hosted: best_effort, optional_on_success

### PR DESCRIPTION
**Pull request by psturc**
* add option to skip hypershift-hostedcluster-dump-hostedcluster step on test success
* add option to ignore failure of hypershift-hostedcluster-destroy-hostedcluster step

### CODE REVIEW

File: ci-operator/step-registry/hypershift/hostedcluster/destroy/hostedcluster/hypershift-hostedcluster-destroy-hostedcluster-ref.yaml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ref:
   as: hypershift-hostedcluster-destroy-hostedcluster
+  best_effort: true
```
Comment (psturc) [inline ci-operator/step-registry/hypershift/hostedcluster/destroy/hostedcluster/hypershift-hostedcluster-destroy-hostedcluster-ref.yaml:3]: https://docs.ci.openshift.org/docs/architecture/step-registry/#marking-post-steps-best-effort

File: ci-operator/step-registry/hypershift/hostedcluster/dump/hostedcluster/hypershift-hostedcluster-dump-hostedcluster-ref.yaml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ref:
   as: hypershift-hostedcluster-dump-hostedcluster
   best_effort: true
+  optional_on_success: true
```
Comment (psturc) [inline ci-operator/step-registry/hypershift/hostedcluster/dump/hostedcluster/hypershift-hostedcluster-dump-hostedcluster-ref.yaml:4]: https://docs.ci.openshift.org/docs/architecture/step-registry/#skipping-post-steps-on-success

**Comment by danilo-gemoli** [comment]
I doesn't seem hard to review but I'm afraid I could miss something...
/cc @bear-redhat

**Pull request merged**